src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -102,56 +107,92 @@ public void update()
         }
     }
 
+    public boolean checkInWareHouse(final BuildingHome buildingEntry, final boolean addToList)
+    {
+        if (buildingEntry.isFoodNeeded())
+        {
+            if (isInHut(itemStack -> !InventoryUtils.isItemStackEmpty(itemStack) && itemStack.getItem() instanceof ItemFood))
+            {
+                if (addToList)
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    taskQueue.add(buildingEntry);
+                }
+                return true;
+            }
+
+            if (taskQueue.contains(buildingEntry))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)